pkg/health/health.go

```diff
@@ -0,0 +1,158 @@
+package health
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/internal/metrics"
+	"github.com/GoogleCloudPlatform/cloud-run-release-operator/pkg/config"
+	"github.com/pkg/errors"
+)
+
+// Diagnosis is the information about the health of the revision.
+type Diagnosis struct {
+	EnoughRequests     bool
+	CheckResults       []*CheckResult
+	FailedCheckResults []*CheckResult
+	IsHealthy          bool
+}
+
+// CheckResult is information about a metrics criteria check.
+type CheckResult struct {
+	MetricsType   config.MetricsCheck
+	MinOrMaxValue interface{}
+	ActualValue   interface{}
+	IsCriteriaMet bool
+	Reason        string
```

Comment:
it seems like MinOrMaxValue (rename: Threshold or ExpectedValue) and ActualValue are what we need in this type.

IsCriteriaMet is ok to.

but the type etc can be inferred from the []config.Metric we passed in.

Reply:
also just use float64